src/auth.c

```diff
@@ -2433,6 +2434,29 @@ size_t pr_auth_set_max_password_len(pool *p, size_t len) {
   return prev_len;
 }
 
+char *pr_auth_bcrypt(pool *p, const char *key, const char *salt,
+    size_t *hashed_len) {
+  char hashed[128], *res;
+
+  if (p == NULL ||
+      key == NULL ||
+      salt == NULL ||
+      hashed_len == NULL) {
+    errno = EINVAL;
+    return NULL;
+  }
+
+  if (bcrypt_hashpass(key, salt, hashed, sizeof(hashed)) != 0) {
```

Comment:
By using the `bcrypt_hashpass` symbol in the core code, we pull that symbol (and all of its underlying Blowfish functions) into the core code by the linker.  And that means that shared modules can find those symbols when they are loaded.  Sneaky.